src/TreeView/TreeView.tsx

```diff
@@ -463,35 +474,17 @@ const Item = React.forwardRef<HTMLElement, TreeViewItemProps>(
               contentVisibility: containIntrinsicSize ? 'auto' : undefined,
               containIntrinsicSize,
             }}
-            onClick={event => {
-              if (onSelect) {
-                onSelect(event)
-              } else {
-                toggle(event)
-              }
-            }}
-            onAuxClick={event => {
-              if (onSelect && event.button === 1) {
-                onSelect(event)
-              }
-            }}
           >
             <div style={{gridArea: 'spacer', display: 'flex'}}>
               <LevelIndicatorLines level={level} />
             </div>
             {hasSubTree ? (
-              // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
               <div
                 className={clsx(
                   'PRIVATE_TreeView-item-toggle',
                   onSelect && 'PRIVATE_TreeView-item-toggle--hover',
                   level === 1 && 'PRIVATE_TreeView-item-toggle--end',
                 )}
-                onClick={event => {
```

Comment:
This might still be needed for the subtree handle to work in the controlled story: https://primer-109b1de022-13348165.drafts.github.io/storybook/?path=/story/components-treeview-features--controlled

https://github.com/primer/react/assets/1863771/3afe6a6c-25f2-47ed-8280-b3641a3cb622

But it is strange that it is a div element 🤔 

Maybe the fix is that this should not be a div or have an explicit role instead?

Reply:
Thanks for looking into this. I was not sure what edge case this handled.
Ideally I'd say it needs to have a `role='button'`? But I'm not sure what do we prefer? Do we want `button` element styled to suit the tree or just add the role to the existing element?

Reply:
> Ideally I'd say it needs to have a role='button'?

That would have been my guess as well, but it looks like that's not the case.

See this [example of treeview from ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/treeview/examples/treeview-navigation/) and the [spec @ericwbailey wrote for our TreeView component](forhttps://github.com/github/primer/blob/main/apis/tree-view-api.md#the-expandcollapse-chevron-toggle)

Reply:
Thanks for the tag here! Tree nodes are a little weird in that they're interaction actions, which might suggest buttons (that's what I originally led with), but it's more a special kind of content unto itself. 

The tree pattern it's mapped to an operating system File Explorer tree, so that's a good mental model to use here. The entire tree itself is focusable, but then individual nodes are not buttons to allow for easier traversal, as well as a less polluted interactive element list if navigating via an element list of similar technique.

Reply:
This structure is definitely tough with `jsx-a11y/no-static-element-interactions` since I think we need a way to distinguish between clicking on the functional label for the treeitem versus both the label and subtree within a `role="treeitem"` 😬 In this case, the container for the label is expecting these events to propagate so that it can do specific behavior when interacting with the label.

In general, this rule feels like it can flag in unexpected ways and the way it suggests or indicates to fix the issue seem to be incorrect 😞 (e.g. suggesting to make the item an interactive role)

Reply:
@siddharthkp - Thanks so much for the link! Really enjoyed reading through such a thorough document.

The spec seems to indicate that the chevron icon exists only to be clicked upon (input pointer only). Keyboard and Screenreaders don't need to know about this interaction. If it was a button, it would be an unnecessary burden for Assistive Technology? This is my interpretation. @ericwbailey 

As for the `jsx-a11y/no-static-element-interactions` rule, I feel like I could build a case for it to be ignored or deprioritised on our accessibility scorecard. I would include `jsx-a11y/click-events-have-key-events` in the list too. WDYT @joshblack ?

Reply:
>If it was a button, it would be an unnecessary burden for Assistive Technology? This is my interpretation. @ericwbailey

Yup, pretty much. The reason behind the approach here is say there's a Treeview with 1,500 files, as well as nested nodes to contain them. Providing buttons here would create 1,500+ tab stops to get through the component.

By making only the Treeview itself focusable and then relying on ARIA announcements and non-<kbd>Tab</kbd> keyboard traversal, we reduce this to one tab stop. This helps out a ton with a lot of different motor control-related accessibility considerations.

Reply:
Ok, so for now I've removed this code change. Elaborated the reasons in the comments and recorded this info in the  issue https://github.com/github/primer/issues/2793